tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -0,0 +1,216 @@
+	return ImageHooksTest::$dimensions ?: getimagesize($filename);
+}
+
+use OCA\MediaMetadata\Services\ImageDimension;
+use OCA\MediaMetadata\Services\ImageDimensionMapper;
+use OCA\MediaMetadata\Tests\TestCase;
+use OCP\ILogger;
+use OCP\IDb;
+use OCP\Files\File;
+use OCP\Files\Folder;
+
+/**
+ * Class ImageHooksTest
+ *
+ * @package OCA\MediaMetadata\Hooks
+ */
+class ImageHooksTest extends TestCase {
+	/**
+	 * @var \OCA\MediaMetadata\Hooks\ImageHooks
+	 */
+	protected $imagehooks;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OC\Files\Node\Root
+	 */
+	protected $root;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ImageDimensionMapper
+	 */
+	protected $mapper;
+	/**
+	 * @var ILogger
+	 */
+	protected $logger;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCP\IDb
+	 */
+	protected $db;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ImageDimension
+	 */
+	protected $imageDimension;
+
+	/**
+	 * @var array $dimensions that will be returned by getimagesize()
+	 */
+	public static $dimensions;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$this->root = $this->getMockBuilder('OC\Files\Node\Root')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->mapper = $this->mockImageDimensionMapper();
+		$this->logger = $this->getMockBuilder('\OCP\ILogger')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->db = $this->getMockBuilder('\OCP\IDb')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->imageDimension = $this->mockImageDimension();
+
+		$app = new \OCA\MediaMetadata\AppInfo\Application();
+		$container = $app->getContainer();
+		$this->imagehooks = new ImageHooks(
+			$this->root,
+			$this->mapper,
+			$container->query('ServerContainer')->getConfig()->getSystemValue('datadirectory')
+		);
+	}
+
+	protected function testPostCreate() {
+		$location = 'testFolder/test.jpg';
+		$fileId = 260495;
+		$jpgFile = $this->mockJpgFile($location, $fileId);
+
+		$imageWidth = 100;
+		$imageHeight = 100;
+		$dimensions = array($imageWidth, $imageHeight);
+
+		$this->mockGetImageSize($location, $dimensions);
+
+		$this->logger->log('debug', 'Test Image Path: '.$location, array('app' => 'MediaMetadata'));
+		$this->logger->log('debug', 'Test Image Height: '.$imageHeight, array('app' => 'MediaMetadata'));
+		$this->logger->log('debug', 'Test Image Width: '.$imageWidth, array('app' => 'MediaMetadata'));
+
+		$this->imageDimension->expects($this->once())
+			->method('setImageId')
+			->with($jpgFile->getId());
+		$this->imageDimension->expects($this->once())
+			->method('setImageHeight')
+			->with($imageHeight);
+		$this->imageDimension->expects($this->once())
+			->method('setImageWidth')
+			->with($imageWidth);
+
+		$this->mapper->expects($this->once())
+			->method('insert')
+			->with($this->imageDimension);
+
+		$this->imagehooks->post_create($jpgFile);
```

Comment:
Ah, that explained why I had to merge then. It would be great if you could rebase the branch.